File: DataBindingStudy/app/src/main/java/com/jie/databinding/CollectionActivity.java
package com.jie.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.util.SparseArray;

import com.jie.databinding.databinding.ActivityCollectionBinding;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class CollectionActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityCollectionBinding collectionBinding = DataBindingUtil.setContentView(this, R.layout.activity_collection);
        String[] strArr = new String[] {"Hi", "Hello"};
        List<String> list = new ArrayList<>();
        list.add("Hello");
        Map<String, String> map = new HashMap<>();
        map.put("jackie", "Hi");
        Set<String> set = new HashSet<>();
        set.add("Hi");
        SparseArray<String> sparseArray = new SparseArray<>();
        sparseArray.put(0, "Hello");
        collectionBinding.setIndex(0);
        collectionBinding.setKey("jackie");
        collectionBinding.setArray(strArr);
        collectionBinding.setList(list);
        collectionBinding.setSet(set);
        collectionBinding.setMap(map);
        collectionBinding.setSparse(sparseArray);
        setTitle("Collection in xml");
    }
}

File: DataBindingStudy/app/src/main/java/com/jie/databinding/ui/activity/CollectionActivity.kt
package com.jie.databinding.ui.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.SparseArray
import androidx.databinding.DataBindingUtil
import com.jie.databinding.R
import com.jie.databinding.databinding.ActivityCollectionBinding

class CollectionActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataBindingUtil.setContentView<ActivityCollectionBinding>(
            this,
            R.layout.activity_collection
        ).apply {
            index = 0
            key = "jackie"
            val strArr = arrayOf("Hi", "Hello")
            val list = mutableListOf("Hello")
            val map = mutableMapOf(Pair<String, String>("jackie", "Hi"))
            val set = mutableSetOf("Hi")
            val sparseArr: SparseArray<String> = SparseArray()
            sparseArr.put(0, "Hello")
            array = strArr
            setList(list)
            setMap(map)
            setSet(set)
            sparse = sparseArr
        }
        title = "Collection in dataBinding study"
    }
}